server/tools/docker-entrypoint.sh

```diff
@@ -62,7 +62,7 @@ fi
 ########################
 
 if [ -z "$BIND" ]; then
-    BIND=$(hostname -i)
+    BIND=$(hostname -i | tr " " "\n" | egrep '([[:digit:]]{1,3}\.){3}[[:digit:]]{1,3}')
```

Comment:
Is it certain that IPv4 addresses would be always available?
Note that only one of the addresses is going to be used since the system property can be set only to one value (iteration on line 70 would pass multiple ones but only a single one would be transferred to system properties)

Reply:
This is a very good question...

So technically, it is possible to use IPv6 with Docker (see [manual](https://docs.docker.com/v17.09/engine/userguide/networking/default_network/ipv6/#using-ndp-proxying)). However, that would require some configuration changes, as `standalone-ha.xml` declares default addresses like that:

```xml
<socket-binding name="jgroups-mping" interface="private" port="0" multicast-address="${jboss.default.multicast.address:230.0.0.4}" multicast-port="45700"/>
<socket-binding name="jgroups-tcp" interface="private" port="7600"/>
<socket-binding name="jgroups-udp" interface="private" port="55200" multicast-address="${jboss.default.multicast.address:230.0.0.4}" multicast-port="45688"/>
```

In addition to that, the `keycloak/bin/standalone.conf` adds this line:

```
keycloak/bin/standalone.conf:   JAVA_OPTS="-Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true"
```

So technically, it should be possible to run Keyclaok (or Wildfly) with IPv6, but it would require major tweaking (supplying `jboss.default.multicast.address` property and `JAVA_OPTS` environmental variable at least). 

Let me ask @belaban @rhusar and @pferraro if they know any IPv6 use cases for Infinispan or Wildfly. According to my knowledge, we use only IPv4 in containers, Kubernetes or OpenShift environment.

Reply:
I think it should be possible to use IPv6 in containers, but most images I know of provide only one (or two) IPv4 addresses.

Reply:
Thanks for the info @belaban! It's similar to what I've seen so far. But it seems @Nowheresly uses a bit more complicated setup.